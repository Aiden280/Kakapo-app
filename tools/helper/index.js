import fs from 'fs-extra';

// Upload stats.json to `http://webpack.github.io/analyse` for analysis.
// Also you can analyze size with `https://github.com/robertknight/webpack-bundle-size-analyzer`:
// `cat stats.json | analyze-bundle-size`
export const writeStats = stats =>
  fs.writeFile('stats.json', JSON.stringify(stats.toJson(), null, 4));

export default {};
