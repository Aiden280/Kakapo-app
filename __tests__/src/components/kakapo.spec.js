import React from 'react';
import { shallow } from 'enzyme';
import { identity, values } from 'ramda';
import {
  ImportKakapo as Kakapo,
  ImportKakapoItem as KakapoItem
} from 'components/';
import { getData, randomSounds } from '../helper';

function setup(props = {}) {
  const propData = {
    ...getData('search'),
    soundActions: {},
    ...getData('intl'),
    dispatch: identity,
    ...props
  };
  return { props, wrapper: shallow(<Kakapo {...propData} />) };
}

test('<Kakapo/> render', () => {
  const { wrapper } = setup();
  expect(wrapper.type()).toBe('div');
  expect(wrapper.prop('className')).toBe('kakapo');
});

test('<Kakapo/> render items', () => {
  const { wrapper } = setup({
    search: { kakapofavs: values(randomSounds(5)) }
  });
  expect(wrapper.find(KakapoItem).length).toBe(5);
});
