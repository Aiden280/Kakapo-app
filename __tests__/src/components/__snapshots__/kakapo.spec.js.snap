// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Kakapo/> render 1`] = `
<div
  className="kakapo"
>
  <h5>
    Add a sound from Kakapo.com
  </h5>
  <div
    className=""
  />
</div>
`;

exports[`<Kakapo/> render items 1`] = `
<div
  className="kakapo"
>
  <h5>
    Add a sound from Kakapo.com
  </h5>
  <div
    className="kakapofavs-items"
  >
    <div
      className="kakapo-item"
      onClick={[Function]}
      role="link"
      tabIndex={-1}
    >
      <div
        className="thumbnail"
      >
        <i
          className="icon-null"
        />
      </div>
      <span
        className="title"
      >
        Thunder
      </span>
    </div>
    <div
      className="kakapo-item"
      onClick={[Function]}
      role="link"
      tabIndex={-1}
    >
      <div
        className="thumbnail"
      >
        <i
          className="icon-null"
        />
      </div>
      <span
        className="title"
      >
        Thunder
      </span>
    </div>
    <div
      className="kakapo-item"
      onClick={[Function]}
      role="link"
      tabIndex={-1}
    >
      <div
        className="thumbnail"
      >
        <i
          className="icon-null"
        />
      </div>
      <span
        className="title"
      >
        Thunder
      </span>
    </div>
    <div
      className="kakapo-item"
      onClick={[Function]}
      role="link"
      tabIndex={-1}
    >
      <div
        className="thumbnail"
      >
        <i
          className="icon-null"
        />
      </div>
      <span
        className="title"
      >
        Thunder
      </span>
    </div>
    <div
      className="kakapo-item"
      onClick={[Function]}
      role="link"
      tabIndex={-1}
    >
      <div
        className="thumbnail"
      >
        <i
          className="icon-null"
        />
      </div>
      <span
        className="title"
      >
        Thunder
      </span>
    </div>
  </div>
</div>
`;
